Doc/library/bisect.rst

```diff
@@ -21,21 +21,27 @@ example of the algorithm (the boundary conditions are already right!).
 The following functions are provided:
 
 
-.. function:: bisect_left(a, x, lo=0, hi=len(a))
+.. function:: bisect_left(a, x, lo=0, hi=len(a), key=None)
 
    Locate the insertion point for *x* in *a* to maintain sorted order.
    The parameters *lo* and *hi* may be used to specify a subset of the list
-   which should be considered; by default the entire list is used.  If *x* is
-   already present in *a*, the insertion point will be before (to the left of)
-   any existing entries.  The return value is suitable for use as the first
-   parameter to ``list.insert()`` assuming that *a* is already sorted.
+   which should be considered; by default the entire list is used. The
+   parameter *key* specifies a function of one argument that is used to
+   extract a comparison key from each element in *a* (for example,
```

Comment:
Having read the C implementation, `key` does more than strictly this, and I can think of one example where that's surprising.

Specifically, `key` is called both on elements in `a` and on `x`.


Sometimes it's all the same: `left(["a", "C", "d"], "b", key=str.lower)`

Sometimes it's obvious/expected: `left(["a", "C", "d"], "B", key=str.lower)`

And for some user that's unexpected: `left([(1, 2), (3, 4, 5), ...], 42, key=sum)`

The last case can be summarised by key function having different argument and return types. I see it that `bisect.left` can be used for 2 purposes — to find an index to insert an element; or to find a value (or None) in the "immutable" array. If someone uses `bisect` for the latter, then `key(x)` may be unexpected.

Please consider somehow including the fact that `key` will be called on `x` too in the documentation.